extension-manifest-v3/src/background/custom-filters.js

```diff
@@ -31,51 +123,88 @@ async function updateDNRRules(dnrRules) {
         id: 1000000 + index,
       })),
     });
+
+    console.info(`Custom Filters: DNR updated with rules: ${dnrRules.length}`);
   }
 }
 
-chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
-  if (msg.action === 'customFilters:engine') {
-    console.groupCollapsed(
-      'Custom filters: generating filters for provided input',
-    );
-    console.log(msg.filters);
-    console.groupEnd();
-
-    try {
-      const {
-        cosmeticFilters,
-        networkFilters,
-        preprocessors,
-        notSupportedFilters,
-      } = parseFilters(msg.filters);
-
-      engines.createEngine(engines.CUSTOM_ENGINE, {
-        cosmeticFilters,
-        networkFilters,
-        preprocessors,
-      });
-
-      sendResponse(
-        notSupportedFilters.map(
-          (msg) => `Filter not supported: '${msg.filter}'`,
-        ),
-      );
-    } catch (e) {
-      sendResponse([e]);
-    }
+function updateEngine(text) {
+  const {
+    networkFilters,
+    cosmeticFilters,
+    preprocessors,
+    notSupportedFilters,
+  } = parseFilters(text);
 
-    return false;
-  }
+  engines.createEngine(engines.CUSTOM_ENGINE, {
+    cosmeticFilters,
+    networkFilters,
+    preprocessors,
+  });
+
+  console.info(
+    `Custom Filters: Engine updated with network filters: ${networkFilters.length}, cosmetic filters: ${cosmeticFilters.length}`,
+  );
+
+  return {
+    networkFilters: networkFilters.length,
+    cosmeticFilters: cosmeticFilters.length,
+    errors: notSupportedFilters.map(
+      (msg) => `Filter not supported: '${msg.filter}'`,
+    ),
+  };
+}
+async function update(text, { trustedScriptlets }) {
+  const parseResult = parseInput(text, { trustedScriptlets });
```

Comment:
I tried to move the logic without changes, but I am not sure if this is still required - as for the engine, the input of the filters is parsed again, this time by the `parseFilters` function from the adblocker.

I left it here for now, as the internal parser uses the `trustedScriptlets` flag - but there was a bug before, which did not pass "fixed" scriptlets to the engine, but it generated them again.